Add ScanOrganic schema 1.0 (#230)
EDCD/EDDN: schemas/scanorganic-README.md

properly structure it for sending to EDDN.

Please consult [EDDN Schemas README](./README-EDDN-schemas.md) for general
documentation for a schema such as this.

If you find any discrepancies between what this document says and what is
defined in the relevant Schema file, then you should, in the first instance,
assume that it is the Schema file that is correct.
**PLEASE open
[an issue on GitHub](https://github.com/EDCD/EDDN/issues/new/choose)
to report any such anomalies you find so that we can check and resolve the
discrepancy.**

## Senders
The primary data source for this schema is the ED Journal event `ScanOrganic`.

### ScanType
The 'Analyse' scan type only triggers when you have the scan tool out for a
long enough duration following your third scan. It's possible to put away the
tool before this completes and the event will then trigger the next time the
tool is used, which could be in another system entirely. For this reason,
it may report incorrect data and is thus excluded from submission.

### Variant
Variant should be reported if present. It was not included in older journal
versions and therefore is not required to facilitate older journal submissions.

> 

Reviewer: Suggestion:

### Use of status.json
You are encouraged to augment your submission with values from status.json. Only status.json values from a live game session may be used. When status.json is used:
- You must verify that the timestamp from status.json is no more than one minute older than the timestamp from the `ScanOrganic` event 
- You must verify that the status.json `Flags2` value `0x00000010` is true (indicating that the commander is on-foot on a planet)
- You must verify that the `BodyName` from status.json and journal events are in agreement
- Please augment your submission with `BodyName`, `Latitude`, and `Longitude` values

> 

Reviewer: Suggestion:

### Use of status.json
You are encouraged to augment your submission with values from status.json. When status.json is used:
- You must verify that the timestamp from status.json is no more than one minute older than the timestamp from the `ScanOrganic` journal event 
- You must verify that the status.json `Flags2` value `0x00000010` is true (indicating that the commander is on-foot on a planet)
- You must verify that the `BodyName` from status.json and journal events are in agreement
- Please augment your submission with `BodyName`, `Latitude`, and `Longitude` values

renamed to BodyID to mirror many other events.

You MUST track `BodyName` both from Status.json *and* also from some
Journal events in order to cross-check it before using the `Body` from
Journal events.

The following is correct as of game version 4.0.0.801 (Odyssey initial
release, Update 7, plus one patch).

1. Record `journal_body_name` and `journal_body_id` from the
   `BodyName` and `BodyID` values in `ApproachBody` events.

   This will occur when the player flies below Orbital Cruise altitude
   around a body.
2. Also record these from `Location` events to cover logging in already there.
3. Unset both `journal_body_name` and `journal_body_id` on `LeaveBody` and
   `FSDJump` events.
   Do NOT do so for `SupercruiseEntry`, because a player can enter supercruise
   below max Orbital Cruise altitude and then come back down without a new
   `ApproachBody` event occurring.
4. If Status.json has `BodyName` present, record that as `status_body_name`.

   This key and its value will be present whenever the player comes close
   enough to a body for the Orbital Cruise/Glide HUD elements to appear.
   It will disappear again when they fly back above that altitude, or jump
   away.

> 5. If Status.json does **not** have `BodyName` then clear `status_body_name`.

Reviewer: I think `BodyName` would need to be an optional augmentation because it requires a cross check using status.json. This value should only be reported when status.json values from a live game session are available.

#### Latitude / Longitude
While latitude and longitude are not reported with the event data, this event
necessitates being on a planet surface. Pulling the current coordinates from
the `Status.json` should be sufficient to populate the data. The `Status.json`

> granularity is quite accurate when on-foot.

Reviewer: I think `Latitude` and `Longitude` would need to be an optional augmentation because it requires using status.json. These values should only be reported when status.json values from a live game session are available.

Please consult [EDDN Schemas README](./README-EDDN-schemas.md) for general
documentation for a schema such as this.

If you find any discrepancies between what this document says and what is
defined in the relevant Schema file, then you should, in the first instance,
assume that it is the Schema file that is correct.
**PLEASE open
[an issue on GitHub](https://github.com/EDCD/EDDN/issues/new/choose)
to report any such anomalies you find so that we can check and resolve the
discrepancy.**

## Senders
The primary data source for this schema is the ED Journal event `ScanOrganic`.

### ScanType
The 'Analyse' scan type only triggers when you have the scan tool out for a
long enough duration following your third scan. It's possible to put away the
tool before this completes and the event will then trigger the next time the
tool is used, which could be in another system entirely. For this reason,
it may report incorrect data and is thus excluded from submission.

### Variant
Variant should be reported if present. It was not included in older journal
versions and therefore is not required to facilitate older journal submissions.

### Use of status.json

> You are encouraged to augment your submission with values from status.json. When status.json is used:

Reviewer: ```suggestion
You are encouraged to augment your submission with latitude and longitude values from status.json. 
```

* You must verify that the timestamp from status.json is no more than one minute older than the timestamp from the `ScanOrganic` journal event
* You must verify that the status.json `Flags2` value `0x00000010` is true (indicating that the commander is on-foot on a planet)
* You must verify that the `BodyName` from status.json and journal events are in agreement

> * Please augment your submission with `BodyName`, `Latitude`, and `Longitude` values

Reviewer: * Please augment your submission with `BodyName`, `StarSystem`, `StarPos`, and (when available) `Latitude`, and `Longitude` values

> * Please augment your submission with `BodyName`, `StarSystem`, `StarPos`, and (when available) `Latitude` and `Longitude` values

Reviewer: Delete lines 35-38?

### Augmentations
#### horizons and odyssey flags
Please read [horizons and odyssey flags](../docs/Developers.md#horizons-and-odyssey-flags)
in the Developers' documentation.

#### gameversion and gamebuild
You **MUST** always set these as per [the relevant section](../docs/Developers.md#gameversions-and-gamebuild)
of the Developers' documentation.

#### StarSystem
You MUST add a StarSystem key/value pair representing the name of the system
this event occurred in. Source this from either Location, FSDJump or
CarrierJump as appropriate.

**You MUST apply a location cross-check, as per
[Other data augmentations](../docs/Developers.md#other-data-augmentations).**

#### StarPos
You MUST add a `StarPos` array containing the system co-ordinates from the
last `FSDJump`, `CarrierJump`, or `Location` event.

**You MUST apply a location cross-check, as per

> [Other data augmentations](../docs/Developers.md#other-data-augmentations).**

Reviewer: These highlighted lines describe augments which are not currently allowed per the schema?

#### BodyID and BodyName
BodyID is already present in the form of the 'Body' key. This should be
renamed to BodyID to mirror many other events.

If proper synchronicity can be achieved, `BodyName` should be reported,
be it from Status.json or from some Journal events. Please cross-check it

> as possible before using the `Body` from Journal events.

Reviewer: ```suggestion
as possible.
```

The following is correct as of game version 4.0.0.801 (Odyssey initial
release, Update 7, plus one patch).

1. Record `journal_body_name` and `journal_body_id` from the
   `BodyName` and `BodyID` values in `ApproachBody` events.

   This will occur when the player flies below Orbital Cruise altitude
   around a body.
2. Also record these from `Location` events to cover logging in already there.
3. Unset both `journal_body_name` and `journal_body_id` on `LeaveBody` and
   `FSDJump` events.
   Do NOT do so for `SupercruiseEntry`, because a player can enter supercruise
   below max Orbital Cruise altitude and then come back down without a new
   `ApproachBody` event occurring.
4. If Status.json has `BodyName` present, record that as `status_body_name`.

   This key and its value will be present whenever the player comes close
   enough to a body for the Orbital Cruise/Glide HUD elements to appear.
   It will disappear again when they fly back above that altitude, or jump
   away.
5. If Status.json does **not** have `BodyName` then clear `status_body_name`.

> 

Reviewer: Remove the data accumulation process in lines 72-93?
